### ISSUE TITLE
sigSegv in renumberMesh

### ISSUE THREAD

**Issue by mrklein** [mrklein/openfoam-os-x#3]
renumberMesh halts with the following error:

```
#0  "Foam::error::printStack(Foam::Ostream&)" in ~/OpenFOAM/OpenFOAM-2.3.1/platforms/darwin64ClangDPOpt/lib/libOpenFOAM.dylib
#1  "Foam::sigSegv::sigHandler(int)" in ~/OpenFOAM/OpenFOAM-2.3.1/platforms/darwin64ClangDPOpt/lib/libOpenFOAM.dylib
#2  "_sigtramp" in /usr/lib/system/libsystem_platform.dylib
#3  ? in /usr/lib/system/libsystem_platform.dylib
#4  "reorderMesh(Foam::polyMesh&, Foam::List<int> const&, Foam::List<int> const&)" in ~/OpenFOAM/OpenFOAM-2.3.1/platforms/darwin64ClangDPOpt/bin/renumberMesh
#5  "main" in ~/OpenFOAM/OpenFOAM-2.3.1/platforms/darwin64ClangDPOpt/bin/renumberMesh
#6  "start" in /usr/lib/system/libdyld.dylib
```

The error seems to be happening in Opt build (since in Debug build of 2.4.x error does not happen).

**Comment by mrklein** [comment]
Reproduced in 2.2.x, 2.3.1, 2.4.0, and 2.4.x. Yet there is no such error in 2.1.x ;)

**Comment by mrklein** [comment]
Resolved by 95b6c271cc9f24e074dbf3b60f1bbeab46fc8bcf
